Issue 23607: first pass of changes for compatibility with Python3

marmeladema opened the pull request:
As much as i want to migrate entirely to Python3 (see #23607), it will require some time as changes in web-platform-tests are significant and rely on upstream fixes to be merged and synced downstream.
In the meantime, lets improve compatibility with Python3 so that later, migration will be less painful.

Build system is definitely not ready yet for Python3, but its a step in the right direction.

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x ] `./mach build -d` does not report any errors
- [ x] `./mach test-tidy` does not report any errors

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

marmeladema committed:
Improve print statement compatibility with Python3

marmeladema committed:
Add six module to python requirements.txt for Python3 compatibility

marmeladema committed:
Use urllib from six module in order to be compatible with Python3

marmeladema committed:
Fix except statement in order to be compatible with Python3

marmeladema committed:
Use octal literal syntax compatible with Python3

marmeladema committed:
Use iteritems from six module for Python3 compatibility

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/servo/mutation/mutator.py, python/servo/devenv_commands.py, python/servo/lints/wpt_lint.py, python/servo/bootstrap.py, python/requirements.txt and 7 more

marmeladema commented:
`AppVeyor` tests are failing because of some rustup issue. Not sure if its really related to my changes?

CYBAI commented:
```
Checking files for tidiness...

[94m./python/servo/command_base.py[0m:[93m355[0m: [91mE502 the backslash is redundant between brackets[0m
```

marmeladema commented:
> ```
> Checking files for tidiness...
> 
> �[94m./python/servo/command_base.py�[0m:�[93m355�[0m: �[91mE502 the backslash is redundant between brackets�[0m
> ```

Should be fixed now!

marmeladema committed:
Fix StringIO module import to be compatible with Python3

jdm commented:
@bors-servo r+
Thanks for working in this!

bors-servo commented:
:pushpin: Commit b90ab44 has been approved by `jdm`

<!-- @bors-servo r=jdm b90ab445c7d57387ba8e5859fb74885f44cf088c -->

bors-servo commented:
:hourglass: Testing commit b90ab445c7d57387ba8e5859fb74885f44cf088c with merge 9f0ac66eb225fb19de860eef6a5954b8e8188d9c...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/13787)

CYBAI commented:
<details>

```
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-3:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
STDERR: Exception in thread TestRunnerManager-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-4:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-6:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-10:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-22:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-23:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-24:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
STDERR: Exception in thread TestRunnerManager-20:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-16:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
STDERR: Exception in thread TestRunnerManager-14:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
STDERR: Exception in thread TestRunnerManager-5:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
STDERR: Exception in thread TestRunnerManager-7:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-21:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-18:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-11:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
STDERR: Exception in thread TestRunnerManager-12:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-19:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
STDERR: Exception in thread TestRunnerManager-8:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
Traceback (most recent call last):
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent

STDERR: Exception in thread TestRunnerManager-17:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 386, in run
    new_state = self.wait_event()
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 461, in wait_event
    return f(*data)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 592, in test_ended
    stack=result.stack)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 51, in inner
    data = converter.convert(*args, **kwargs)
  File "/home/servo/buildbot/slave/linux-rel-wpt/build/python/_virtualenv/lib/python2.7/site-packages/mozlog/logtypes.py", line 96, in convert
    raise TypeError("Unrecognised argument %s" % key)
TypeError: Unrecognised argument known_intermittent
```

</details>

marmeladema commented:
I think I will need help to fix the failing test because i can't manage to reproduce it locally with `./mach test-wpt`
Moreover the problem seems to happen in mozlog python package called from wptrunner, both components I have not touched.
I did notice in `python/requirements.txt` that `mozlog == 3.6` is specified whereas `tests/wpt/web-platform-tests/tools/wptrunner/requirements.txt` specify `mozlog==4.2.0` but i don't know if its related at all.

marmeladema commented:
I rebased over current master and was able to reproduce the issue, but as soon as I deleted `python/_virtualenv/` so that it gets rebuild at the next call of `./mach` it worked and I was able to pass the tests again.

CYBAI commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit bf66a08c023e0f8fe93ad1a41dae90e0ca12c002 with merge 28211734177c349b563b7c67b038509610d35ca3...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/13794)

jdm commented:
Oh, those Linux machines will have an existing virtualenv. We should modify the buildbot_steps.yml to include a command to clean it

marmeladema committed:
Remove python/_virtualenv/ dir in clean_build_artifacts.sh

marmeladema commented:
@jdm i added a line in `clean_build_artifacts.sh` to remove the virtualenv but maybe you would prefer that to be an explicit command in `buildbot_steps.yml`?

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 2b3f6a2 has been approved by `jdm`

<!-- @bors-servo r=jdm 2b3f6a228929a0297f82fff18bfffa98707bb031 -->

bors-servo commented:
:hourglass: Testing commit 2b3f6a228929a0297f82fff18bfffa98707bb031 with merge 6d488f1be24c1b679931d6d02703f4a10759eb49...

bors-servo commented:
:sunny: Test successful - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/13741), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/13806), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/UpS4TPAjSfWHzj6_gxy1kQ)
Approved by: jdm
Pushing 6d488f1be24c1b679931d6d02703f4a10759eb49 to master...
